lib/ansible/modules/cloud/azure/azure_rm_acr.py

```diff
@@ -0,0 +1,362 @@
+        default: resource_group location
+        required: false
+    admin_user_enabled:
+        description:
+            - If enabled, you can use the registry name as username and admin user access key as password to docker login to your container registry.
+        default: false
+        required: false
+    sku:
+        description:
+            - Specifies the SKU to use. Currently can be either Basic, Standard or Premium.
+        default: Standard
+        choices:
+            - Basic
+            - Standard
+            - Premium
+        required: false
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Yawei Wang (@yaweiw)"
+
+'''
+
+EXAMPLES = '''
+    - name: Create an azure container registry
+      azure_rm_acr:
+        name: testacr1
+        location: eastus
+        resource_group: testrg
+        state: present
+        admin_user_enabled: true
+        sku: Premium
+        tags:
+            Release: beta1
+            Environment: Production
+
+# Deletes the specified container registry in the specified subscription and resource group.
+    - name: Remove an azure container registry
+      azure_rm_acr:
+        name: testacr2
+        resource_group: testrg
+        state: absent
+'''
+RETURN = '''
+state:
+    description: Current state of the azure container registry and it's credentials
+    returned: always
+    type: dict
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.containerregistry.models import (
+        Registry,
+        RegistryUpdateParameters,
+        StorageAccountProperties,
+        Sku,
+        SkuName,
+        SkuTier,
+        ProvisioningState,
+        PasswordName,
+        WebhookCreateParameters,
+        WebhookUpdateParameters,
+        WebhookAction,
+        WebhookStatus
+    )
+    from azure.mgmt.containerregistry import ContainerRegistryManagementClient
+except ImportError as exc:
+    # This is handled in azure_rm_common
+    pass
+
+
+def create_acr_dict(registry, credentials):
+    '''
+    Helper method to deserialize a ContainerRegistry to a dict
+    :param: registry: return ACR object from Azure rest API call
+    :param: credentials: return credential objects from Azure rest API call
+    :return: dict of return ACR and it's credentials
+    '''
+    results = dict(
+        id=registry.id if registry is not None else "",
+        name=registry.name if registry is not None else "",
+        location=registry.location if registry is not None else "",
+        admin_user_enabled=registry.admin_user_enabled if registry is not None else "",
+        sku=registry.sku.name if registry is not None else "",
+        provisioning_state=registry.provisioning_state if registry is not None else "",
+        password_name=credentials.passwords[0].name.value if credentials is not None else "",
```

Comment:
Might be better to map the returned credentials to a sub-dict named `credentials` whose keys are the `name` fields and values are the passwords.